app/services/csv_preview_service.rb

```diff
@@ -43,7 +47,9 @@ def newline_or_last_char_index(string, newline_index)
   end
 
   def truncate_to_maximum_number_of_lines(string, maximum_number_of_lines)
-    string[0..newline_or_last_char_index(string, maximum_number_of_lines - 1)]
+    truncation_index = newline_or_last_char_index(string, maximum_number_of_lines - 1)
```

Comment:
I wonder if we could add an early return to this method? something like this above this line maybe?
```
return string unless string.length > maximum_number_of_lines
```